google/cloud/spanner_dbapi/connection.py

```diff
@@ -187,15 +229,16 @@ def _rerun_previous_statements(self):
                     self.connection._transaction = None
                     raise Aborted(status.details)
 
-                retried_checksum = ResultsChecksum()
-                retried_checksum.consume_result(res)
-                retried_checksum.consume_result(status.code)
+                if not self.read_only:
```

Comment:
This should not be necessary. The _rerun_previous_statements method should never be called when we are in read-only mode (and also not when in auto-commit mode).

Re-running a list of statements is only necessary when we are in a multi-use read/write transaction.